contracts/src/v0.8/automation/dev/v2_3/AutomationRegistrar2_3.sol

```diff
@@ -164,104 +173,33 @@ contract AutomationRegistrar2_3 is TypeAndVersionInterface, ConfirmedOwner, IERC
 
   //EXTERNAL
 
-  /**
-   * @notice register can only be called through transferAndCall on LINK contract
-   * @param name string of the upkeep to be registered
-   * @param encryptedEmail email address of upkeep contact
-   * @param upkeepContract address to perform upkeep on
-   * @param gasLimit amount of gas to provide the target contract when performing upkeep
-   * @param adminAddress address to cancel upkeep and withdraw remaining funds
-   * @param triggerType the type of trigger for the upkeep
-   * @param checkData data passed to the contract when checking for upkeep
-   * @param triggerConfig the config for the trigger
-   * @param offchainConfig offchainConfig for upkeep in bytes
-   * @param amount quantity of LINK upkeep is funded with (specified in Juels)
-   * @param sender address of the sender making the request
-   */
-  function register(
```

Comment:
From the desc in L#14, this register function is entrance used by UI. A few questions:
- If we get rid of this, the UI needs to use `registerUpkeep` in L#180? 
- this `registerUpkeep` in L#180 needs to be payable to take native? 
- for `amount` and  `billingToken` in `RegistrationParams `,  is it possible that users have `msg.value != 0` and `billingToken != address(0)`? I guess we can figure that from UI?  However, if that happens, for our solidity check, should we take it as Native payment or ERC20 payment?

Reply:
this function was essentially a helper function for the `onTokenTransfer()` function to call into. If you look at the modifier, you see it was only ever callable by the link token, so no one should really be affected by this change. We just got rid of an unnecessary middleman.

Good question on `msg.value` - I think that will be addresses in a follow up PR, since these functions are currently not `payable`.